NetworkingSampleApp/NetworkingSampleApp/Scenes/Users/UsersViewModel.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  UsersViewModel.swift
+//  NetworkingSampleApp
+//
+//  Created by Matej Molnár on 07.12.2023.
+//
+
+import Foundation
+import Networking
+
+@MainActor
+final class UsersViewModel: ObservableObject {
+    @Published var users = [User]()
+    @Published var createdUser: SampleCreateUserResponse?
+    
+    /// Custom decoder needed for decoding `createdAt` parameter of SampleCreateUserResponse.
+    private let responseDecoder: JSONDecoder = {
+        let decoder = JSONDecoder()
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"
+        decoder.dateDecodingStrategy = .formatted(dateFormatter)
+        return decoder
+    }()
+
+    private lazy var apiManager: APIManager = {
+        var responseProcessors: [ResponseProcessing] = [
+            LoggingInterceptor.shared,
+            StatusCodeProcessor.shared
+        ]
+        var errorProcessors: [ErrorProcessing] = [LoggingInterceptor.shared]
+
+#if DEBUG
+        responseProcessors.append(EndpointRequestStorageProcessor.shared)
+        errorProcessors.append(EndpointRequestStorageProcessor.shared)
+#endif
+
+        return APIManager(
+            requestAdapters: [LoggingInterceptor.shared],
+            responseProcessors: responseProcessors,
+            errorProcessors: errorProcessors
+        )
+    }()
+}
+
+extension UsersViewModel {
+    func getUsers(in range: ClosedRange<Int>, parallelFetch: Bool) {
+        Task {
+            users = []
```

Comment:
do we need to reset users like this? or is it different from users.removeAll()? Do we need to call it in task or can it be before the task starts?

Reply:
No it doesn't matter whether we do it inside the Task or with the `removeAll()` method. What is your preference?

Reply:
No difference in performance between those two btw as long as you don't wish to keep the original capacity.